src/subprocess-posix.cc

```diff
@@ -184,6 +209,8 @@ SubprocessSet::SubprocessSet() {
     Fatal("sigaction: %s", strerror(errno));
   if (sigaction(SIGTERM, &act, &old_term_act_) < 0)
     Fatal("sigaction: %s", strerror(errno));
+  if (sigaction(SIGCHLD, &act, &old_chld_act_) < 0)
```

Comment:
I think setting a shared handler with sigterm and sigint is dangerous, as SIGCHLD gets an opportunity to override value of interrupted_ global (which could be set to e.g., SIGINT), preventing ninja from exiting promptly.